### ISSUE TITLE
Cario install problem -- undefined symbol: cairo_svg_surface_set_document_unit

### ISSUE THREAD

**Issue by Moribund7** [pygobject/pycairo#148]
Hi,
I have problem with installing pycairo. I try follow instruction https://pycairo.readthedocs.io/en/latest/getting_started.html
but while trying import cairo i get this message:

Python 3.7.2 (default, Dec 29 2018, 06:19:36) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cairo
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/filip/anaconda3/lib/python3.7/site-packages/cairo/__init__.py", line 1, in <module>
    from ._cairo import *  # noqa: F401,F403
ImportError: /home/filip/anaconda3/lib/python3.7/site-packages/cairo/_cairo.cpython-37m-x86_64-linux-gnu.so: undefined symbol: cairo_svg_surface_set_document_unit

Installation end without any error. 

Thanks for any help,
Filip

**Comment by lazka** [comment]
You somehow have an older cairo version now compared to when you build pycairo.

**Comment by Moribund7** [comment]
I try uninstall pycairo and cairo and install it again, but still get the same error.

**Comment by lazka** [comment]
you have to pass `--no-cache-dir` to `pip install` otherwise it will just install it again as is.

**Comment by lazka** [comment]
I don't know any way to tell pip to not cache build results and re-use them in a different environment. So not much we can do here -> closing.

**Comment by shivam1423** [comment]
Hi I'm also facing the similar issue.


import cairo
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/shivam/anaconda3/lib/python3.7/site-packages/cairo/__init__.py", line 1, in <module>
    from ._cairo import *  # noqa: F401,F403
ImportError: /home/shivam/anaconda3/lib/python3.7/site-packages/cairo/_cairo.cpython-37m-x86_64-linux-gnu.so: undefined symbol: cairo_svg_surface_set_document_unit

I working on ubantu with pycairo-1.19.1
Is there any other solution?

**Comment by matanster** [comment]
I'm having the same issue. One of the above comments is suggesting that the cairo version installed is incompatible for building the most recent pycairo. However, installing pycairo per-se goes fine:

```script
pip install pycairo --no-cache-dir
```

It's while going through the installation of python gtk that this problem arises:

```script
pip install PyGObject --no-cache-dir
```

Maybe someone might be able to kindly say what I am not getting about it ...

**Comment by Jwolter0** [comment]
> Here is a workaround posted elsewhere:
> https://www.reddit.com/r/Python/comments/doq8ye/installing_pycairo_issue/

Same problem, and that worked for me.

TL;DR: try 

pip3 install pycairo==1.11.1

**Comment by TianyiShi2001** [comment]
> TL;DR: try
> 
> pip3 install pycairo==1.11.1

This works but doesn't really solve the problem. A newer version should be expected to be back-compatible.

**Comment by naveen521kk** [comment]
Try
```sh
pip install pycairo --no-cache --force --no-binary :all:
````

**Comment by tamhanepd** [comment]
> Try
> 
> ```shell
> pip install pycairo --no-cache --force --no-binary :all:
> ```

It worked for me. PyGObject wasn't getting installed despite having Cairo. Using this command and replacing `pycairo` with `pygobject` worked.

**Comment by americapat** [comment]
unresolved for me:
$ pip3 install PyGObject --no-cache --force --no-binary :all:
Collecting PyGObject
  Downloading PyGObject-3.42.0.tar.gz (716 kB)
...
Successfully built pycairo
Failed to build PyGObject
ImportError: /tmp/pip-build-env-0m6jvsct/overlay/lib/python3.7/site-packages/cairo/_cairo.cpython-37m-x86_64-linux-gnu.so: undefined symbol: cairo_svg_surface_set_document_unit
Much appreciated if someone can figure this one, happy to provide more info. -- Pat.

**Comment by Neltherion** [comment]
@americapat I have the same problem as you. Did you find any solutions?

**Comment by Hubert-YH** [comment]
same problems， I tried all the suggestions yet still failed

**Comment by Hubert-YH** [comment]
> > TL;DR: try
> > pip3 install pycairo==1.11.1
> 
> This works but doesn't really solve the problem. A newer version should be expected to be back-compatible.

I think I have got a possible answer, at least for me, some functions should be achieved both with Cairo and pyoject, yet the least version requires a version newer than 1.16 but perhaps the newest version of Cairo did not match PyObject well, and it might already abandon symbols like  cairo_tee_surface_index, but some old software is still using it. ideally, all these stuff should be arranged simultaneously, but it turns out to be different in the reality

**Comment by Hubert-YH** [comment]
In my practice, it is from Cairo-1.14.0, that cairo_svg_surface_set_document_unit was deleted
from changelog we can see:
1.14.0 - 2017-07-12
General:
Requires at least cairo 1.13.1 (The snapshop in Ubuntu 14.04)

Tests:
Optional Hypothesis tests.

New Features:
Add Surface.set_device_scale() and Surface.get_device_scale(). #pr-44 (Sander Sweers)

Add Device #pr-45

Add Surface.get_device() #pr-45

Add ScriptDevice and ScriptMode #pr-46

Add ScriptSurface #17

Add Status.JBIG2_GLOBAL_MISSING

Add Format.stride_for_width()

Add TextClusterFlags and SurfaceObserverMode

Add Gradient.get_color_stops_rgba()

Add TeeSurface !! here !!
Add MeshPattern

Add TeeSurface !! here must be related to cairo_tee_surface_index or other cairo tee stuffs
